File: app/src/main/java/hyunwook/co/kr/paging_livedata/viewmodel/WordViewModel.java
package hyunwook.co.kr.paging_livedata.viewmodel;

import android.app.Application;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import hyunwook.co.kr.paging_livedata.entity.Word;
import hyunwook.co.kr.paging_livedata.repository.WordRepository;

import java.util.List;

public class WordViewModel extends AndroidViewModel {

    private WordRepository mRepository;
    private LiveData<List<Word>> mAllWords;

    public WordViewModel(Application application) {
        super(application);
        mRepository = new WordRepository(application);
        mAllWords = mRepository.getAllWords();
    }

    public LiveData<List<Word>> getAllWords() {
        return mAllWords;
    }

    public void insert(Word word) {
        mRepository.insert(word);
    }

}

File: app/src/main/java/hyunwook/co/kr/paging_livedata/viewmodel/WordViewModel.kt
package hyunwook.co.kr.paging_livedata.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import hyunwook.co.kr.paging_livedata.entity.Word
import hyunwook.co.kr.paging_livedata.repository.WordRepository

class WordViewModel constructor(application: Application): AndroidViewModel(application) {

    private val mRepository = WordRepository(application)
    val allWords: LiveData<List<Word>> = mRepository.getAllWords()

    fun insert(word: Word) {
        mRepository.insert(word)
    }
}